ls1intum/Artemis
# `Adaptive learning`: Improve participation in learning path view

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


This PR addresses improvements related to participation in exercises and lecture units using Learning Paths in Artemis.
Stacked on #6747 .

### Description

- fixes several style issues
- adds generation of LP if a student's LP is missing and requested for the first time

### Steps for Testing

Prerequisites:
- 1 Admin
- 1 Instructor
- 1 Students/User (not enrolled in the course)
- All Exercises and a Lecture with all lecture units excluding exercise unit
- One Competency that contains all the exercises and lecture units

1. Log in to Artemis as Instructor
2. Navigate to Learning Path Management and enable learning paths
3. Log in to Artemis as Admin
4. Manually add the Student/User to the student group of the course
5. The List of Learning Paths in `Learning Path Management` of the course should not contain the Student's Learning Path
6. Log in to Artemis as Student/User
7. Navigate to the Courses Learning Path tab
8. Log in to Artemis as Instructor
9. Navigate to Learning Path Management and make sure you can see the student's learning path in the list
10. Log in to Artemis as Student and navigate to the Courses Learning Path tab
For each exercise/lecture unit
11. Click on the `Next` button to open a lecture unit/exercise
12. Participate in the lecture unit/exercise and make sure it works as expected
13. Also check the conversations tab on the right side (correct channel? Is it possible to post & react?)
14. Return to the learning path and continue (11.) until there are no new recommendations

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage







### Screenshots


<img width="1719" alt="Screenshot 2023-08-01 at 12 55 58" src="https://github.com/ls1intum/Artemis/assets/44003963/ac19fccc-9e81-49dc-8417-070742f369f8">
